### PR TITLE
Fix 6862

### LINKED ISSUE #6862: amazon: With no credentials gives misleading error.
```
$ packer build -connection-info -only amazon-ebs cloud.json                                                                                
amazon-ebs output will be in this color.

1 error(s) occurred:

* Unknown region: eu-west-1
```

### PR DESCRIPTION
**Pull request by SwampDragons**
Make sure we expose errors with amazon credentials rather than burying those underneath a region validation error. 

Closes #6862

TODO update tests

### CODE REVIEW

File: builder/amazon/common/regions.go [hunk 1]
```diff
@@ -15,23 +15,31 @@ func getValidationSession() *ec2.EC2 {
...
 
-	return regions
+	return regions, nil
 }
 
 // ValidateRegion returns true if the supplied region is a valid AWS
 // region and false if it's not.
-func ValidateRegion(region string, ec2conn ec2iface.EC2API) bool {
-	for _, valid := range listEC2Regions(ec2conn) {
+func ValidateRegion(region string, ec2conn ec2iface.EC2API) (bool, error) {
```
Comment (azr) [inline builder/amazon/common/regions.go]: Here, I believe the bool is not necessary anymore. When the region is invalid, then an error saying why should be returned.

[PR 6787](https://github.com/hashicorp/packer/pull/6787/files#diff-743b70f36793a4f1edd68908b6048ef1) had something like this.

May be we could cherry pick those commit so we don't have to wait.

**Comment by SwampDragons** [comment]
Good point. I'll rewrite to just use the error.

**Comment by SwampDragons** [comment]
done!

**Comment by azr** [comment]
Awesome 🙂

**Pull request merged**